src/runtime/recipe/internal/handle.ts

```diff
@@ -256,6 +256,17 @@ export class Handle implements Comparable<Handle>, PublicHandle {
     return this._capabilities;
   }
 
+  private hasHardRef(connection: HandleConnection): boolean {
+    if (connection.type && connection.type.getEntitySchema()) {
+      for (const field of Object.values(connection.type.getEntitySchema().fields)) {
+        if (field.annotations.some(a => a.name === 'hardRef')) {
```

Comment:
is it possible that one of the fields is nested and a nested field has the hardRef annotation?

Reply:
right, will handle in a follow up pr